contribution-and-developer-guidelines.md

```diff
@@ -0,0 +1,102 @@
+# CoatRack Contribution and Developer Guidelines
+
+
+
+## Introduction
+
+#### Possibilities to Contribute
+
+Hello, dear prospective contributor. You can support CoatRack in many ways:
+
+- Report bugs using GitHub's issue tracker.
+- Discuss the current state of the code in open Pull Requests.
+- Submit a bugfix.
+- Propose new features and improvement ideas.
+- Improve the documentation.
+- Clean up code.
+- Add tests.
+- Help us to fulfill the [FIWARE requirements](https://fiware-requirements.readthedocs.io/en/latest/).
+
+
+
+Note: If you want to contribute to CoatRack you need to sign the corresponding Contributors License Agreement (CLA) previously. Signing instructions can be found in [signing-tutorial.md](https://github.com/coatrack/signing-tutorial.md). 
+
+
+
+## Conventions
+
+#### Applying Best Practices
+
+The establishment of best practices in CoatRack and compliance of these is important to us. We put great effort to our review process to steadily improve the quality of our code base and we expect the same willingness from you. Here are a few practices that should always be considered: 
+
+* Code is written for readers and should be quickly understood by them. Use descriptive names, a clear architecture, short functions and small, cohesive classes. Add Java Docs if necessary.
+* Use a consistent coding style that goes along with the already existing code. Please do not change the formatting of exiting code that is not relevant to the issue you are working on.
+* Spaces should be used for indentation, not tabs.
+* Newly added code should be delivered with automated tests, for example unit tests or integration tests. Ensure to have a high code coverage.
+
+
+
+#### Workflow
+
+We use GitHub to host code, track issues and feature requests, as well as accept pull requests. We use the following workflow:
+
+1. Fork the repo. Of course, this step must be performed only once. All others are run again for each issue.
```

Comment:
Should we also refer here that the user/contributor should disable the worflow that tries to push the code of the master branch to dev.

Reply:
If that is still required, we should add it here.

Reply:
Done. The above mentioned phrase suggested by @BrunoSilvaATB was added.